src/db.c

```diff
@@ -1587,350 +1587,349 @@ static void _db_node_mask_fixup(struct db_arg_chain_tree *node)
-		 *        A       |  +------------------+  |   |
-		 *        C     FALSE                     TRUE |
-		 *        C  <----+                        |   |
-		 *        E  <----+           +------------+   |
-		 *        P     FALSE         v                |
-		 *        T       |  +------------------+      |
-		 *                +--|  Al(x) >= Rl(x)  |------+
-		 *                   +------------------+
-		 *
-		 * SCMP_CMP_LE:
-		 *                   +------------------+
-		 *                +--|  Ah(x) >  Rh(x)  |------+
-		 *                |  +------------------+      |
-		 *              FALSE                         TRUE     R
-		 *                |                            |       E
-		 *                +-----------+                +---->  J
-		 *                            v                +---->  E
-		 *                   +------------------+      |       C
-		 *                +--|  Ah(x) == Rh(x)  |--+   |       T
-		 *        A       |  +------------------+  |   |
-		 *        C     FALSE                     TRUE |
-		 *        C  <----+                        |   |
-		 *        E  <----+           +------------+   |
-		 *        P     FALSE         v                |
-		 *        T       |  +------------------+      |
-		 *                +--|  Al(x) >  Rl(x)  |------+
-		 *                   +------------------+
-		 *
-		 * SCMP_CMP_EQ:
-		 *                   +------------------+
-		 *                +--|  Ah(x) == Rh(x)  |--+
-		 *        R       |  +------------------+  |           A
-		 *        E     FALSE                     TRUE         C
-		 *        J  <----+                        |           C
-		 *        E  <----+           +------------+   +---->  E
-		 *        C     FALSE         v                |       P
-		 *        T       |  +------------------+      |       T
-		 *                +--|  Al(x) == Rl(x)  |------+
-		 *                   +------------------+
-		 *
-		 * SCMP_CMP_NE:
-		 *                   +------------------+
-		 *                +--|  Ah(x) == Rh(x)  |--+
-		 *        A       |  +------------------+  |           R
-		 *        C     FALSE                     TRUE         E
-		 *        C  <----+                        |           J
-		 *        E  <----+           +------------+   +---->  E
-		 *        P     FALSE         v                |       C
-		 *        T       |  +------------------+      |       T
-		 *                +--|  Al(x) == Rl(x)  |------+
-		 *                   +------------------+
-		 *
-		 */
-
-		/* setup the level */
-		switch (chain[iter].op) {
+	arg = api_arg->arg;
+	mask = api_arg->mask;
+	datum = api_arg->datum;
+
+	/* link this level to the previous level */
+	for (iter = 0; prev_nodes[iter] != NULL; iter++)
+		if (*tf_flag)
+			prev_nodes[iter]->nxt_t = _db_node_get(c_iter[0]);
+		else
+			prev_nodes[iter]->nxt_f = _db_node_get(c_iter[0]);
+	prev_nodes[0] = prev_nodes[1] = prev_nodes[2] = prev_nodes[3] = NULL;
```

Comment:
Personal nit-pick: please don't chain assignments together like this, put them each on their own line.